### PR TITLE
test(toolkit): pin FieldState contracts and submitWithWarnings edge cases

### PR DESCRIPTION
**Pull request by the-ult**
## Summary

> ⚠️ **Stack**: this PR targets `fix/toolkit-warning-classification` (#110), not `main`. It must be merged *after* #110 (or rebased onto `main` after that merge). The drift tests assert behaviors introduced in that branch (WeakSet in-flight guard, `markAsTouched()` delegation).

- **`field-state-types.spec.ts`** (new): runtime contract assertions for `ErrorVisibilityState` and `ErrorReadableState` — the two `Pick<FieldState, …>` types in the toolkit's public API. Builds a real Angular form in `TestBed` and asserts the duck-typed contract (`invalid()`/`touched()` return booleans, `errors()` returns an array of `{kind: string}` objects). Also includes `expectTypeOf` type-level documentation (inert until spec typecheck debt is paid — see PR #111 note). If an Angular major-version bump renames these members, `pnpm nx test toolkit` turns red at the toolkit boundary instead of in consumer apps.
- **`submission-helpers.drift.spec.ts`** (extended, 4→6 scenarios):
  - Rejected action: `submitWithWarnings` re-throws (matches Angular 22 native `submit()` behavior); form is re-submittable after (WeakSet cleared in `finally`).
  - Async validator pending: pins the `pending()` guard — action is not invoked while the async validator is still in flight; after resolution with a valid result, the action runs.

## Test plan

- [ ] `pnpm nx test toolkit` — 1168 tests pass (up from 1153 on `main`)
- [ ] `git status` shows no production `.ts` files modified

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Added contract tests validating type exports and runtime behavior for form field state management.
  * Enhanced form submission handler test coverage for action rejection handling and pending async validator resolution scenarios.

### CODE REVIEW

File: packages/toolkit/core/utilities/submission-helpers.drift.spec.ts [hunk 1]
```diff
@@ -201,4 +202,126 @@ describe('submitWithWarnings — Angular submit() drift guard', () => {
...
+    // effect() + resource() wiring). Do NOT await whenStable() here — that
+    // would block forever because the resource loader is parked on the gate.
+    await Promise.resolve();
+    await Promise.resolve();
+    await Promise.resolve();
+
+    // The form should be pending — the async validator's resource is loading.
+    expect(asyncForm().pending()).toBe(true);
+
+    const action = vi.fn(async () => {});
```
[the remark this answers is not in the document]
Comment (the-ult) [reply packages/toolkit/core/utilities/submission-helpers.drift.spec.ts]: Good call — replaced the fixed three-microtask wait with `vi.waitFor(() => expect(asyncForm().pending()).toBe(true))`. It polls instead of assuming a tick count, and still avoids `whenStable()` (which would hang on the gated loader). Fixed in 74a20468.

File: packages/toolkit/core/utilities/submission-helpers.drift.spec.ts [hunk 2]
```diff
@@ -201,4 +203,126 @@ describe('submitWithWarnings — Angular submit() drift guard', () => {
...
+    await expect(
+      submit(nativeForm, async () => {
+        await rejectingAction();
+        return null;
+      }),
+    ).rejects.toThrow('server error');
+    expect(rejectingAction).toHaveBeenCalledOnce();
+    rejectingAction.mockClear();
+
+    // submitWithWarnings re-throws the rejection — diverges from native submit().
```
[the remark this answers is not in the document]
Comment (the-ult) [reply packages/toolkit/core/utilities/submission-helpers.drift.spec.ts]: Addressed — the comment now reads "matches native submit() on this dimension" rather than implying divergence (already in the current head, d36cd303).
